### PR TITLE
Routing and Connecting Sign in and Sign Up

### PR DESCRIPTION
**Pull request by farahwamer**
Set up routing and connected sign in and sign up. Now you have to be signed in to access any of the routes, and you can only access the pages that are available to you based on your user role.

http://localhost:3000/ should now automatically take you to the sign in page.

1. Create an account, choose whether you want to create an admin or user account.
2. Use the account to sign in.
3. You can use adminer or the API docs to create loans associated with your account to see your signed out items show up on your signed out page.
4. Settings page should be populated with the signed in user's data

**HEADS UP:** the logout button is in the settings page :)

**ANOTHER HEADS UP**:  the input fields were still buggy when i was doing this PR, so I _removed all regex checking on the inputs_ so that they work normally.

### CODE REVIEW

**Comment by manda-02** [comment]
something to note: when I sign up as a User, it automatically makes me an admin until I went into the Adminer and edited the account

File: client/src/pages/SignUp.tsx [hunk 1]
```diff
@@ -26,34 +64,74 @@ const SignUp: React.FC<SignUpProps> = ({ onSignIn }) => {
...
+                    }
+                  />{" "}
                 </div>
                 <div className="py-8">
-                  <InputField label="Password" />
+                  <InputField
+                    label="Password"
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                      setPassword1(e.target.value)
+                    }
```
Comment (victortarn) [inline client/src/pages/SignUp.tsx:108]: Should be password type so password cannot be viewed while user is typing

File: client/src/pages/SignUp.tsx [hunk 2]
```diff
@@ -26,34 +64,74 @@ const SignUp: React.FC<SignUpProps> = ({ onSignIn }) => {
...
+                    onClick={handleSignUp}
                   >
-                    Sign In
+                    Sign Up
                   </button>
                 </div>
               </div>
             </div>
             <div className="ml-8 pt-8">
               <span>Don't already have an account?</span>
```
Comment (victortarn) [inline client/src/pages/SignUp.tsx:130]: Should read: "Already have an account?

File: client/src/pages/SignUp.tsx (hunk 1 above)
Comment (farahwamer) [reply client/src/pages/SignUp.tsx:109]: nice catch!

File: client/src/pages/SignUp.tsx (hunk 2 above)
Comment (farahwamer) [reply client/src/pages/SignUp.tsx]: Fixed!

**Review by victortarn — APPROVED** [review]

**Review by manda-02 — APPROVED** [review]
user sign up now works! well done!

**Pull request merged**